Automattic/vip-go-mu-plugins
# Add new subsite wp-blogs-update CLI command

## Description
This PR adds a `wp vip subsites wp-blogs-update` command, purpose of which is updating the `domain` and `path` of wp_blogs. We are adding this command because Search and Replace doesn't cover wp_blogs in a good way:

### wp_blogs search and replace problem

Consider the following wp_blogs structure:

```
+---------+----------+----------+
| blog_id | domain   | path     |
+---------+----------+----------+
|       1 | example.com | subsite1 |
+---------+----------+----------+
|       2 | example.com | subsite2 |
+---------+----------+----------+
```

If we want to change subsite with blog_id 2 from `example.com/subsite2` to `subsite2.example.com`, we cannot just simply use search and replace because:
- if we do search and replace from `example.com/subsite2` to `subsite2.example.com` nothing will happen
- if we do search and replace from `example.com` to `subsite2.example.com` all entries will get changed

The above could also be achieved by using existing `wp db` CLI commands, but we wanted to add this as a `wp vip` CLI command, in case we'll need to extend the functionality or add more commands.

## Changelog Description

### New VIP CLI command: Update domain and path of a subsite in wp_blogs

We have added a new `wp vip subsites update-wp-blogs` command, which updates the domain and path values for a subsite in wp_blogs.

## Checklist

Please make sure the items below have been covered before requesting a review:

- n\a This change has relevant unit tests (if applicable).

## Steps to Test

1. Check out PR.
2. Setup a test multisite environment
3. Run the CLI command: `wp vip subsites update-wp-blogs <blog-id> <domain> --subsite-path=<path>`
4. Verify that `domain`, `path` and `last_updated` values are changed in wp_blogs for the provided blog-id
